File: c/fizzbuzz.c
#include <stdio.h>

int fizzbuzz(int n, int max) {
    if (n % 3 == 0 && n % 5 == 0 && printf("FizzBuzz\n")) {} 
    else if (n % 3 == 0 && printf("Fizz\n")) {}
    else if (n % 5 == 0 && printf("Buzz\n")) {}
    else if (printf("%d\n", n)) {}

    if (n < max && fizzbuzz(n + 1, max)) {}
}

int main() {
    if (fizzbuzz(1, 100)) {}
}

File: rust/fizzbuzz.rs
fn main() {
  // you can either call it like this
  if fizzbuzz(1, 100) == () {}

  // OR like this. This one triggers a warning, though.
  //if let _ = fizzbuzz(1, 100) {}
}

fn fizzbuzz(num: u32, max: u32) {
    if let mut s = String::new() {
        if num % 5 == 0 && s.push_str("Fizz") == () {}
        if num % 3 == 0 && s.push_str("Buzz") == () {}
        if s.is_empty() && s.push_str(&num.to_string()) == () {}

        if println!("{}", s) == () {}

        if num < max && fizzbuzz(num + 1, max) == () {}
    }
}